### PR TITLE
Cleanup nginx config

### PR DESCRIPTION
**Pull request by davidrapson**
Builds on from https://github.com/biglotteryfund/blf-alpha/pull/2825 and cleans up and documents our nginx configs.

### CODE REVIEW

File: deploy/nginx.conf [hunk 1]
```diff
@@ -1,10 +1,9 @@
+# Directives in this file are loaded in the global `http` block
+
 # Passenger config
 client_max_body_size 50M;
 passenger_core_file_descriptor_ulimit 50000;
 
-# Switch on gzip for static assets
-gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;
```
Comment (davidrapson) [inline deploy/nginx.conf]: This is set in `server.conf` so we don't need this twice.

File: deploy/server.conf [hunk 2]
```diff
@@ -1,46 +1,48 @@
...
+    # user_routes is defined in nginx.conf
     location ~ ^/user {
         limit_req zone=user_routes burst=20;
     }
 
     location ~ ^/welsh/user {
         limit_req zone=user_routes burst=20;
     }
-
-    # Media: images, icons, video, audio, HTC
```
Comment (davidrapson) [inline deploy/server.conf]: Given that everything under `/assets` is sent directly to S3 this block should never get executed.

Comment (mattandrews) [reply deploy/server.conf]: True, I guess this only gets called when accessing the site directly, but we can live with that not being cached etc (everything else isn't anyway) 👍

Comment (davidrapson) [reply deploy/server.conf]: Good point, happy to keep this in to account for that use-case?

**Pull request closed without merging**